web-registry/src/hooks/useQueryBaskets.ts

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from 'react';
+
+import {
+  QueryBasketsResponse,
+  QueryClientImpl,
+} from '@regen-network/api/lib/generated/regen/ecocredit/basket/v1/query';
+
+import { useLedger } from '../ledger';
+
+export default function useQueryBaskets(): QueryBasketsResponse | undefined {
+  const { api } = useLedger();
+  const [baskets, setBaskets] = useState<QueryBasketsResponse | undefined>();
+
+  useEffect(() => {
+    if (!api?.queryClient) return;
+
+    const queryClient: QueryClientImpl = new QueryClientImpl(api.queryClient);
+    queryClient
+      .Baskets({})
+      .then(setBaskets)
+      /* eslint-disable */
+      .catch(console.error);
+  }, [api?.queryClient]);
```

Comment:
I wasn't aware you could use optional chaining in effect dependencies! I expected this to error (i've run into this before and had to extract the optionally-chained value to a variable and use that for the dep array), but it seems to be working for me. Maybe that was a typescript bug that React fixed. Good to know!